cmake/extensions.cmake

```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Set 'domain_out' to domain for given board
```

Comment:
I would suggest some more information for the people that only reads the code
On face value i don't understand what domain or domain_out means in this context

Reply:
Great spot, I will add some explanation here, and in the partition_manager.cmake.

There should be some explanation in the .rst, so please check that as well to see if what is stated there makes sense.

Reply:
Renamed to get_pm_domain, and added following doc:
```
# Get the Partition Manager 'domain' for a given board.
#
# In a multi core/SoC build, each core/SoC makes out a domain as seen from the
# Partition Manager. This function removes any 'ns' suffix from the board name
# so that the correct name of the domain remains.
#
# This is needed since TrustZone information is encoded in the board name,
# but the domain is not affected by this TrustZone property (secure or
# non-secure).
#

```

Reply:
Added this to `partition_manager.cmake`

```
# Get the domain of the current board.
# In a multi core/SoC build, each core/SoC makes out a domain as seen from the
# Partition Manager. Each domain is handled individually by the Partition
# Manager. Once the Partiton Manager has solved all the domains, it creates
# a global configuration consisting of all domain configurations.
get_pm_domain(${BOARD} domain)
```

Reply:
And also this to `partition_manager.cmake`

```
# Check if current image is the dynamic partition in its domain.
# I.E. it is the only partition without a statically configured size.
# This is equivalent to the 'app' partition in the root domain.
if("${IMAGE_NAME}" STREQUAL "${${domain}_PM_DOMAIN_DYNAMIC_PARTITION}_")
  set(is_dynamic_partition_in_domain TRUE)
endif()
```

Reply:
Great! I would change "makes out a domain" -> "is a domain"